docs/tracingpolicy/README.md

```diff
@@ -0,0 +1,1193 @@
+and will make it return `-1` every time the first argument is equal to the
+string `/etc/passwd`:
+
+```yaml
+kprobes:
+- call: "__x64_sys_symlinkat"
+  syscall: true
+  args:
+  - index: 0
+    type: "string"
+  - index: 1
+    type: "int"
+  - index: 2
+    type: "string"
+  selectors:
+  - matchArgs:
+    - index: 0
+      operator: "Equal"
+      values:
+      - "/etc/passwd\0"
+    matchActions:
+    - action: Override
+      argError: -1
+```
+
+Note that `Override` can override the return value of any call but doing so in
+kernel functions can create unexpected code path execution. While syscall are a
+stable user interface that should handle errors gracefully.
+
+##### FollowFD action
+
+The `FollowFD` action allows to create a mapping using a BPF map between file
+descriptors numbers and filenames. It however needs to maintain a state
+correctly, see [`UnfollowFD`](#unfollowfd-action) and
+[`CopyFD`](#copyfd-action) related actions.
+
+The `fd_install` kernel function  is called each time a file descriptor must be
+installed into the file descriptor table of a process, typically referenced
+within system calls like `open` or `openat`. It is a good place for tracking
+file descriptor and filename matching.
+
+Let's take a look at the following example:
+```yaml
+- call: "fd_install"
+  syscall: false
+  args:
+  - index: 0
+    type: int
+  - index: 1
+    type: "file"
+  selectors:
+  - matchPIDs:
+      # [...]
+    matchArgs:
+      # [...]
+    matchActions:
+    - action: FollowFD
+      argFd: 0
+      argName: 1
+```
+
+This action uses the dedicated `argFd` and `argName` fields to get respectively
+the index of the file descriptor argument and the index of the name argument in
+the call.
+
+##### UnfollowFD action
+
+The `UnfollowFD` action takes a file descriptor from a system call and deletes
+the corresponding entry from the BPF map, where it was put under the `FollowFD`
+action.
+
+Let's take a look at the following example:
+```yaml
+- call: "__x64_sys_close"
+  syscall: true
+  args:
+  - index: 0
+    type: "int"
+  selectors:
+  - matchPIDs:
+    - operator: NotIn
+      followForks: true
+      isNamespacePID: true
+      values:
+      - 0
+      - 1
+    matchActions:
+    - action: UnfollowFD
+      argFd: 0
+```
+
+Similar to the `FollowFD` action, the index of the file descriptor is described
+under `argFd`:
+```yaml
+matchActions:
+- action: UnfollowFD
+  argFd: 0
+```
+
+In this example, `argFD` is 0. So, the argument from the `__x64_sys_close`
+system call at `index: 0` will be deleted from the BPF map whenever a
+`__x64_sys_close` is executed.
+```yaml
+- index: 0
+  type: "int"
+```
+
+Note, that whenever we would like to follow a file descriptor with a `FollowFD`
+block, there should be a matching `UnfollowFD` block, otherwise the BPF map
+will be broken.
+
+##### CopyFD action
+
+The `CopyFD` action is specific to duplication of file descriptor use cases.
```

Comment:
What is the use case for this? Can we include 2 sentence?

Reply:
I think there is no particular use case for `CopyFD` apart from making the whole `FollowFD` story works. Like you need this to handle the `dup` syscall family but that's not particularly interesting in itself.

Reply:
What do you mean exactly under "making the whole FollowFD story works"?

Reply:
I mean that it's just part of the feature that allows you to follow the file descriptor and path association. `CopyFD` is just a particular thing to handle the `dup` syscalls but is not that interesting in itself. In order to have a proper following of file descriptor/path association, we should `FollowFD` on `install_fd`, try to `UnfollowFD` on all the code parts that are able to close file descriptors, and use `CopyFD` on all the parts that are duplicating file descriptor for the same path.